### PR TITLE
FIX update operator usage in create entity cases

### PR DESCRIPTION
**Pull request by fgalan**
Hardening limitations in issue #3814 

Pending:

- [x] full .test
- [x] CNR
- [x] doc fixes

### CODE REVIEW

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 1]
```diff
@@ -4407,7 +4415,6 @@ unsigned int processContextElement
         /* Successful creation: send potential notifications */
         std::map<std::string, TriggeredSubscription*>  subsToNotify;
         std::vector<std::string>                       attrNames;
-        std::vector<std::string>                       attributes;
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp:4410]: Not related with this PR, but I have realized this variable is not needed.

NTC (informative)

File: test/functionalTest/cases/1112_metadata_as_keymap/metadata_with_dots_v1.test [hunk 2]
```diff
@@ -171,7 +171,7 @@ Content-Length: 269
 
 03. Check the '=' in DB
 =======================
-{ "attrs" : { "A" : { "type" : "T", "creDate" : REGEX(.*), "value" : "foo", "md" : { "m=x" : { "type" : "string", "value" : "black" }, "m=y" : { "type" : "string", "value" : "sweet" } }, "mdNames" : [ "m.x", "m.y" ] } } }
+{ "attrs" : { "A" : { "value" : "foo", "type" : "T", "creDate" : REGEX(.*), "md" : { "m=x" : { "type" : "string", "value" : "black" }, "m=y" : { "type" : "string", "value" : "sweet" } }, "mdNames" : [ "m.x", "m.y" ] } } }
```
Comment (fgalan) [inline test/functionalTest/cases/1112_metadata_as_keymap/metadata_with_dots_v1.test:174]: This change (and the other in the _v2.test file) are due to the changes in the moment in which valueBson() is called in src/lib/mongoBackend/MongoCommonUpdate.cpp. Now `value` is calculated before than `type`, so this change. However, it doesn't have any actual impact (as we used the usual semantic in JSON under which the key order is not relevant).

NTC (just informative)

File: doc/manuals/orion-api.md [hunk 3]
```diff
@@ -1629,41 +1628,16 @@ So be careful of avoiding these situations.
...
-}
-```
-
-then C will be created with value `2`.
+* Numeric operators takes 0 as reference. For instance, `{"$inc": 4}` results in 4,
+  `{$mul: 1000}` results in 0, etc.
+* `$set` takes the empty object (`{}`) as reference. For instance, `"$set": {"X": 1}` results in just `{"X": 1}`
+* `$push` and `$addToSet` take the empty array (`[]`) as reference. For instance, `{"$push": 4}`
+  results in `[ 4 ]`.
+* `$pull`, `$pullAll` and `$unset` are ignored.
```
Comment (mapedraza) [inline doc/manuals/orion-api.md:1640]:
```suggestion
* Attributes containing `$pull`, `$pullAll` and `$unset` operators are ignored. For instance, creating an entity with 2 attributes, the first one containing an operator `"a":{"value":{"$pull":[1]}, ... }"` and `"b":{"value":3...}`, just a normal one, will result in an entity with just one attribute, `b`.
```

Comment (fgalan) [reply doc/manuals/orion-api.md]: Fixed in e4c61c4df

Comment (mapedraza) [reply doc/manuals/orion-api.md]: Ok

**Review by mapedraza — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
@fisuda this PR does some modifications to orion-api.md. It would be great if you could sync in the Japanese translation, please. Thanks!

**Pull request merged**